library/src/lib/calendar/calendar-views/calendar-day-view/calendar-day-view.component.ts

```diff
@@ -0,0 +1,521 @@
+
+        // The calendar grid can have 6 (42 days) weeks
+        const nextMonthDisplayedDays = 42 - calendarDays.length;
+
+        for (let nextD = 1; nextD <= nextMonthDisplayedDays; nextD++) {
+            const fdDate = new FdDate(year, month, nextD);
+            calendarDays.push({ ...this.getDay(fdDate), monthStatus: 'next' });
+        }
+        return calendarDays;
+    }
+
+    /**
+     * Method that generates whole day model basing on fdDate, disabling functions, block functions, and actually
+     * chosen range / single date.
+     * */
+    private getDay(fdDate: FdDate): CalendarDay {
+        const day: CalendarDay = {
+            date: fdDate,
+            weekDay: fdDate.getDay(),
+            disabled: this.disableFunction(fdDate),
+            blocked: this.blockFunction(fdDate),
+            selected: (
+                (this.calType === 'single' && CalendarService.datesEqual(fdDate, this.selectedDate)) ||
+                (this.selectedRangeDate && CalendarService.datesEqual(fdDate, this.selectedRangeDate.start)) ||
+                (this.selectedRangeDate && CalendarService.datesEqual(fdDate, this.selectedRangeDate.end))
+            ),
+            selectedFirst: (this.selectedRangeDate && CalendarService.datesEqual(fdDate, this.selectedRangeDate.start)),
+            selectedLast: (this.selectedRangeDate && CalendarService.datesEqual(fdDate, this.selectedRangeDate.end)),
+            selectedRange: (this.selectedRangeDate && (
+                (this.selectedRangeDate.start && (this.selectedRangeDate.start.getTimeStamp() < fdDate.getTimeStamp())) &&
+                (this.selectedRangeDate.end && (this.selectedRangeDate.end.getTimeStamp() > fdDate.getTimeStamp()))
+            )),
+            ariaLabel: this.calendarI18n.getDayAriaLabel(fdDate.toDate())
+        };
+
+        if (this.calType === 'range' && (this.selectCounter === 0 || this.selectCounter === 2)) {
+            if (this.disableRangeStartFunction && !day.disabled) {
+                day.disabled = this.disableRangeStartFunction(day.date);
+            }
+            if (this.blockRangeStartFunction && !day.blocked) {
+                day.blocked = this.blockRangeStartFunction(day.date);
+            }
+        } else if (this.selectCounter === 1) {
+            if (this.disableRangeEndFunction && !day.disabled) {
+                day.disabled = this.disableRangeEndFunction(day.date);
+            }
+
+            if (this.blockRangeEndFunction && !day.blocked) {
+                day.blocked = this.blockRangeEndFunction(day.date);
+            }
+        }
+
+        return day;
+    }
+
+    /** @hidden */
+    private getShortWeekDays(): string[] {
+        return this.calendarI18n.getAllShortWeekdays()
+            .slice(this.startingDayOfWeek - 1)
+            .concat(this.calendarI18n.getAllShortWeekdays().slice(0, this.startingDayOfWeek - 1))
+            .map(weekday => weekday[0].toLocaleUpperCase());
+    }
+}
```

Comment:
In general this file should have more comments in the code to explain what things do